src/usage_metrics/scripts/save_zenodo_metrics.py

```diff
@@ -51,16 +51,27 @@ def save_zenodo_logs() -> pd.DataFrame():
     and then upload to the sources.catalyst.coop GCS bucket.
     """
 
-    def _yell_if_missing_hits(request_json, page_size: int):
-        """Raise an error if we are not getting the correct amount of API hits."""
-        hits = request_json["hits"]["hits"]
-        if len(hits) != (actual_total := request_json["hits"]["total"]):
-            raise AssertionError(
-                f"The API request says there is {actual_total} total records but we got "
-                f"{len(hits)}! The Zenodo API restricts the number of records to 25 without "
-                f"a size, so we are asking for {page_size} records in the page size, "
-                f"but there are {actual_total}. Consider increasing the page_size."
-            )
+    def get_all_records(url: str, page: int = 1, page_size: int = 25):
+        """Iterate through pages and get all records.
+
+        Also return the JSON itself, which is used to maintain the original format
+        of data in cases where we are querying more than one page.
```

Comment:
```suggestion
    def get_all_records(url: str, page: int = 1, page_size: int = 25):
        """Iterate through pages and get all records.

        Zenodo requires authentication for API requests over a page size of 25.
        Instead of navigating authentication, this function grabs all the records
        by iterating through several pages at the page size limit.
        Also return the JSON itself, which is used to maintain the original format
        of data in cases where we are querying more than one page.
```